react-backend/client/src/components/admin/adminUsersPage.js

```diff
@@ -0,0 +1,297 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { Button } from '../visitors/button';
+import { Logoutbutton } from '../visitors/logoutbutton';
+
+export class AdminUsersPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      users: [],
+      reauthenticated: false,
+      failure: false,
+      password: '',
+      activities: [],
+      filters: [],
+      orderBy: '',
+    };
+  }
+
+  headers = new Headers({
+    Authorization: localStorage.getItem('token'),
+    'Content-Type': 'application/json',
+  });
+
+  handleChange = e => {
+    let newState = {};
+    newState[e.target.name] = e.target.value;
+    this.setState(newState);
+  };
+
+  handleFetchError = res => {
+    if (res.status === 500) throw new Error();
+    return res;
+  };
+
+  setErrorMessage = (error, errorString) => {
+    // console.log(error) // Uncomment to display full errors in the console.
+    this.setState({ errorMessage: errorString });
+  };
+
+  updateResults = () => {
+    fetch('/fetchUsersFilteredBy', {
+      method: 'POST',
+      headers: this.headers,
+      body: JSON.stringify({
+        filterBy: this.state.filters,
+        orderBy: this.state.orderBy,
+      }),
+    })
+      .then(res => {
+        if (res.status === 500) {
+          throw new Error('500');
+        } else {
+          return res.json();
+        }
+      })
+      .then(users => {
+        this.setState(users);
+      })
+      .catch(error => {
+        this.props.history.push('/internalServerError');
+      });
+  };
+
+  sort = e => {
+    this.setState(
+      {
+        orderBy: e.target.value,
+      },
+      this.updateResults
+    );
+  };
+
+  filter = (group, e) => {
+    const filterBy = group + '@' + e.target.value;
+    const isAdding = e.target.checked;
+    let newFilters = [...this.state.filters];
```

Comment:
This instead of the following else if:

```js
 const newFilters = const newFilters = isAdding
      ? [...this.state.filters, filterBy]
      : newFilters.filters(filter => filter !== filterBy);
```